shared/macros-bash.jinja

```diff
@@ -39,6 +39,151 @@ populate {{{ name }}}
     }}}
 {{%- endmacro -%}}
 
+{{#
+# Macro to perform remediation for 'audit_rules_privileged_commands' rule
+#
+# Expects two arguments:
+#
+# tool		    tool used to load audit rules
+# 			    either 'auditctl' or 'augenrules'
+# min_auid		minimum original ID the user logged in with
+#
+# Example macro invocation(s):
+#
+#      perform_audit_rules_privileged_commands_remediation "auditctl" "500"
+#      perform_audit_rules_privileged_commands_remediation "augenrules"	"1000"
+#
+#}}
+{{%- macro bash_perform_audit_rules_privileged_commands_remediation(tool, min_auid) -%}}
+files_to_inspect=()
+{{% if tool == "auditctl" %}}
+# If the audit tool is 'auditctl', then:
+# * add '/etc/audit/audit.rules'to the list of files to be inspected,
+# * specify '/etc/audit/audit.rules' as the output audit file, where
+#   missing rules should be inserted
+files_to_inspect=("/etc/audit/audit.rules")
+output_audit_file="/etc/audit/audit.rules"
+{{%- else -%}}
+# If the audit tool is 'augenrules', then:
+# * add '/etc/audit/rules.d/*.rules' to the list of files to be inspected
+#   (split by newline),
+# * specify /etc/audit/rules.d/privileged.rules' as the output file, where
+#   missing rules should be inserted
+readarray -t files_to_inspect < <(find /etc/audit/rules.d -maxdepth 1 -type f -name '*.rules' -print)
+output_audit_file="/etc/audit/rules.d/privileged.rules"
+{{%- endif %}}
```

Comment:
I have a new idea that we could check that it's one of "auditctl" and "augenrules" and produce an error when the macro is used with some different value as a parameter.

Something like:

```
{{% if tool == "auditctl" %}}
...
{{% elif tool == "augenrules" %}}
...
{{%- else -%}}
{{{ raise("Unknown tool used: " + tool) }}}
{{%- endif %}}

```

The `raise()` is already used in other macros in this file so you can get inspired.